compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -817,19 +808,25 @@ public void visit(BLangIf ifStmt) {
         }
         boolean ifStmtReturns = this.statementReturns;
         boolean currentErrorThrown = this.errorThrown;
+        boolean ifWithinTrxMode = this.withinTransactionScope;
         this.resetStatementReturns();
         this.resetErrorThrown();
+        if (prevTxMode) {
+            this.resetWithinTrxScopeToTrue();
+        }
         if (ifStmt.elseStmt != null) {
-            if (independentBlocks) {
-                commitRollbackAllowed = true;
-                withinTransactionScope = true;
-            }
             analyzeNode(ifStmt.elseStmt, env);
-            if ((prevCommitCount != commitCount) || prevRollbackCount != rollbackCount) {
-                commitRollbackAllowed = false;
+            boolean hasVisitedCommitOrRollback = false;
```

Comment:
Can you check this fix instead for the else part against the tests
          
           analyzeNode(ifStmt.elseStmt, env);
            boolean elseWithinTrxMode = this.withinTransactionScope;
            this.statementReturns = ifStmtReturns && this.statementReturns;
            this.errorThrown = currentErrorThrown && this.errorThrown;
            this.withinTransactionScope = (!prevTxMode || ifWithinTrxMode || elseWithinTrxMode) && prevTxMode;

Reply:
updated in 0135c17 . fixed the tests in https://github.com/ballerina-platform/module-ballerinai-transaction/pull/8/commits/cd53c3a2fe5faa1caf8f0015697371e59dfb0dfd